src/snowflake/snowpark/mock/_udtf.py

```diff
@@ -71,36 +71,18 @@ def _do_register_udtf(
         _emit_ast: bool = True,
         **kwargs,
     ) -> UserDefinedTableFunction:
-
-        # Capture original parameters.
-        ast = None
-        if _emit_ast:
+        if "object_name" in kwargs:
             stmt = self._session._ast_batch.assign()
             ast = with_src_position(stmt.expr.udtf, stmt)
+            ast_id = stmt.var_id.bitfield1
 
-            build_udtf(
-                ast,
+            return MockUserDefinedTableFunction(
                 handler,
-                output_schema=output_schema,
```

Comment:
I assume we do not need those args in the mock plan?

Reply:
Yes, the use_ast path isn't relevant for local testing mock.  For the nop plan we only use these fields of the mock UDTF.